code/_core/obj/item/magazine/_magazine.dm

```diff
@@ -21,10 +21,32 @@
 	var/bullet_diameter_max = -1
 
 	var/icon_states = 1
+	var/bluespaced = FALSE
+	var/regenerate = FALSE
+	var/regen_counter = 0
 
 	weight = 0.25
 
 
+/obj/item/magazine/PostInitialize()
+	start_thinking(src)
+	return ..()
+
+/obj/item/magazine/think()
```

Comment:
I'm not sure I like how this is all setup because start_thinking() would be called regardless if it actually needs to think or not, and the start_thinking() should really be used for things that should be thinking really fast. What you should do instead is use the CALLBACK() define instead in a loop. ``/obj/structure/interactive/blob/proc/check_mobs()`` in _blob.dm is a good example on how to do this.


I'm aware that some other objects coded by me are coded like this, but that's bad code and I wish to change it in the future.